.github/workflows/coverage-linux.yml

```diff
@@ -37,11 +37,11 @@ jobs:
     if: github.event.pull_request.draft == false
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
```

Comment:
Does it really make sense for Actions provided by GitHub? Since they're the ones who also provide the hardware, we _have to_ trust them at some point, and keeping this updated seems more likely to happen if we're using a tag rather than a commit hash. +1 one for doing it for the codecov action though. wdyt?

Reply:
I agree with you it's debatable. After all, if GitHub itself gets hijacked we would have bigger problems. About keeping the hash updated it's a good point. Dependabot can take care of updating the hash and the version tag comment, which is more readable. But, yes, keeping the tag version or tag major can be a good option for GitHub actions if you folks prefer.

Reply:
> Dependabot can take care of updating the hash and the version tag comment, which is more readable.

To provide context: https://github.com/dependabot/dependabot-core/issues/4691#issuecomment-1296532496

Reply:
Shall we resolve the linting warning?
I do this pinning also in my personal projects but have the comment above the actual `use` line.

Reply:
Shall we resolve the linting warning?
I do this pinning also in my personal projects but have the comment above the actual `uses` line.

Reply:
In my opinion it is good to pin all actions, also those of GitHub. We can not be sure if and who pushes what code. What if someone has bad intentions and joins GitHub just to get malicious code into such solutions?

Also at least I do not know which persons maintain the project, have push and write permissions and if I can trust every single one.

Some PRs also seem to be spam or malicious: https://github.com/actions/checkout/pulls

Reply:
I believe I fixed the lint warning but kept the version comment inline, cause, in my understanding, dependabot can only update the version comment inline.

Reply:
Perfect, thank you very much. This is ok for me as long as dependabot can automatically handle this.